### PR TITLE
Add `always-on-top` property to Window element

### PR DESCRIPTION
**Pull request by colelawrence**
It's a very simple implementation, which could be made by mostly pattern matching the existing code path for no-frame and no_frame. I've added my own review for things that stood out as potential questions.

### CODE REVIEW

File: internal/backends/qt/qt_window.rs [hunk 1]
```diff
@@ -1567,11 +1568,12 @@ impl WindowAdapterSealed for QtWindow {
             }
         };
 
-        cpp! {unsafe [widget_ptr as "QWidget*",  title as "QString", size as "QSize", background as "QBrush", no_frame as "bool"] {
+        cpp! {unsafe [widget_ptr as "QWidget*",  title as "QString", size as "QSize", background as "QBrush", no_frame as "bool", always_on_top as "bool"] {
             if (size != widget_ptr->size()) {
                 widget_ptr->resize(size.expandedTo({1, 1}));
             }
             widget_ptr->setWindowFlag(Qt::FramelessWindowHint, no_frame);
+            widget_ptr->setWindowFlag(Qt::WindowStaysOnTopHint, always_on_top);
```
Comment (colelawrence) [inline internal/backends/qt/qt_window.rs:1576]: Found in this list:
https://doc.qt.io/qt-6.2/qtwidgets-widgets-windowflags-example.html

File: internal/backends/winit/winitwindowadapter.rs [hunk 2]
```diff
@@ -332,6 +332,11 @@ impl<Renderer: WinitCompatibleRenderer + 'static> WindowAdapterSealed
         winit_window.set_title(&window_item.title());
         winit_window
             .set_decorations(!window_item.no_frame() || winit_window.fullscreen().is_some());
+        winit_window.set_window_level(if window_item.always_on_top() {
+            winit::window::WindowLevel::AlwaysOnTop
+        } else {
+            winit::window::WindowLevel::Normal
+        });
```
Comment (colelawrence) [inline internal/backends/winit/winitwindowadapter.rs:339]: Question: Does being fullscreen matter like it does for `no-frame`?

File: internal/compiler/builtins.slint [hunk 3]
```diff
@@ -152,6 +152,7 @@ component WindowItem {
     in property <brush> color <=> background;
     in property <string> title: "Slint Window";
     in property <bool> no-frame;
+    in property <bool> always-on-top;
```
Comment (colelawrence) [inline internal/compiler/builtins.slint:155]: Is the ordering arbitrary, here? In the docs, the ordering is alphabetical.

File: internal/backends/winit/winitwindowadapter.rs (hunk 2 above)
Comment (tronical) [reply internal/backends/winit/winitwindowadapter.rs:339]: I don't think it matters :)

File: internal/compiler/builtins.slint (hunk 3 above)
Comment (tronical) [reply internal/compiler/builtins.slint:155]: Yeah. Even the alphabetical ordering in the docs I would personally rather group by relevance. This looks good to me.

File: internal/core/items.rs [hunk 4]
```diff
@@ -1174,6 +1174,7 @@ pub struct WindowItem {
     pub background: Property<Brush>,
     pub title: Property<SharedString>,
     pub no_frame: Property<bool>,
+    pub always_on_top: Property<bool>,
```
Comment (tronical) [inline internal/core/items.rs:1177]: There aren't many window items around, so I think this is fine, too.

**Comment by tronical** [comment]
Looks good to me. Let's see what the others say :)

**Comment by ogoffart** [comment]
Very good PR, complete with documentation and changelog entry. There are no test but I can't think how this could be tested. Thank you for the contribution.

**Pull request merged**